File: src/main/java/me/lulu/hypixelutilities/craft/QuickCraftUtils.java
package me.lulu.hypixelutilities.craft;

import lombok.experimental.UtilityClass;
import me.lulu.hypixelutilities.HypixelUtilities;

@UtilityClass
public class QuickCraftUtils {
    public boolean isFavorite(QuickCraft quickCraft) {
        return getConfig().isFavorite(quickCraft);
    }

    public void toggleFavorite(QuickCraft quickCraft) {
        QuickCraftConfig config = getConfig();
        if (config.isFavorite(quickCraft))
            config.removeQuickCraft(quickCraft);
        else
            config.addQuickCraft(quickCraft);
    }

    private QuickCraftConfig getConfig() {
        return HypixelUtilities.QUICK_CRAFT_CONFIG;
    }
}

File: src/main/java/me/lulu/hypixelutilities/craft/QuickCraftUtils.kt
package me.lulu.hypixelutilities.craft

import me.lulu.hypixelutilities.HypixelUtilities


fun QuickCraft.isFavorite(): Boolean {
    return config.isFavorite(this)
}

fun QuickCraft.toggleFavorite() {
    val config = config
    if (isFavorite()) config.removeQuickCraft(this) else config.addQuickCraft(this)
}

private val config: QuickCraftConfig
    private get() = HypixelUtilities.QUICK_CRAFT_CONFIG
